lib/rack/test.rb

```diff
@@ -54,7 +61,7 @@ class Session
       extend Forwardable
       include Rack::Test::Utils
 
-      def self.new(app, default_host = DEFAULT_HOST) # :nodoc:
+      def self.new(app, default_host = Rack::Test.default_host) # :nodoc:
```

Comment:
Just thinking about alternatives, is there no easy way to inject the default host argument on a per-test/spec basis?

What about if we made this `default_host:` options and provided some kind of `app_options` or something similar so you could write:

```
let(:app_options) {{default_host: '...'}}
```

Reply:
I don't think per test/spec is possible.  You can override the URI provided for each request, but the CookieJar always uses the default host for the session, and `Rack::Test::Methods` does not provide a way to set the default_host, without overriding `build_rack_test_session`.

I think it may be better to modify `build_rack_test_session`. If the current scope responds to `default_host`, it calls it and uses it as the second argument.  Otherwise, it continues to pass one argument (`app`).  That way you can allow for global, per-suite, and per-test/spec setting of default_host, with minimal change to the current implementation.  That seems like a better approach to me, what do you think?  If you like that approach, I can implement it.